src/Compilers/CSharp/Portable/Symbols/Source/SourceLocalSymbol.cs

```diff
@@ -76,6 +80,22 @@ internal class SourceLocalSymbol : LocalSymbol
             _valEscapeScope = Binder.ExternalScope;
         }
 
+        private static DeclarationScope GetScope(bool scoped, bool allowRefKind, TypeSyntax typeSyntax)
+        {
+            if (typeSyntax is RefTypeSyntax refTypeSyntax)
+            {
+                if (refTypeSyntax.ScopedKeyword.Kind() == SyntaxKind.ScopedKeyword)
+                {
+                    return DeclarationScope.ValueScoped;
+                }
+                if (allowRefKind && scoped)
+                {
```

Comment:
It might be good if we could somehow hint that this `scoped` flag is about whether a scoped appeared to the left of the `ref`, i.e. in the "primary" modifiers list. Not sure whether/how to do that.